src/libs/notifications/notifications.ts

```diff
@@ -110,6 +110,59 @@ export const handleServerNotificationOpening = (
   })
 }
 
+/**
+ * Called twice when receiving a notification generated by `showLocalNotification`
+ *
+ * First when the notification is displayed
+ * Second when the notificatin is clicked
+ *
+ * This is called when the user clicks on a notificaiton whatever if the app is in
+ * Foreground, in Background or if the app is not running
+ *
+ * @param client - CozyClient instance
+ * @returns Method to unregister the handler
+ */
+export const handleLocalNotificationOpening = (
+  client: CozyClient
+): (() => void) => {
+  return notifee.onForegroundEvent((event: Event) => {
+    // If no pressAction this mean the handler has been triggered
+    // by `showLocalNotification` but the user did not clicked on it yet
+    if (!event.detail.pressAction) return
+
+    log.debug('Received notification from notifee.onForegroundEvent event')
+
+    void handleNotificationEvent(client, event.detail.notification?.data)
+  })
+}
+
+/**
+ * When receiving a server notification while the App is in Foreground
+ * then no OS notification toast is displayed
+ *
+ * Instead the App will directly receive the notification message
+ *
+ * We want this message to be redirected as a displayed toast
+ *
+ * @returns Method to unregister the handler
+ */
+export const handleServerNotificationOnForeground = (): (() => void) => {
+  return messaging().onMessage(async notification => {
+    log.debug('Received notification from onMessage event')
+
+    if (!notification.notification?.title || !notification.notification.body) {
+      log.error('Received a notification without title or body. Skip.')
```

Comment:
log.error will go to sentry no ? Is it really needed ?

Reply:
Yes. This case should never happen, and if this happens I suppose we should warn the dev team using Sentry.

Reply:
Don't you think that one day it will ? Like tomorrow product team may decide that we want a notification with just a title ? And then every user who do not update will trigger the log.